Hunting Queries/Microsoft 365 Defender/Command and Control/SuspiciousNetworkConnection-Googleapis.yaml

```diff
@@ -0,0 +1,18 @@
+id: fe4f4336-a2e7-4cae-af43-3210106475be
+name: Suspicious network connection to Google APIs - Possible GC2-Sheet C2
+description: |
+  This query will look for connections made to several Google cloud service API endpoints which are not associated with a legitimate browser or application. The query was inspired by the GC2-Sheet project: https://github.com/looCiprian/GC2-sheet. GC2-Sheet is a C2 platform designed to leverage Google Sheets and Google Drive SaaS applications.
+requiredDataConnectors:
+- connectorId: MicrosoftThreatProtection
+  dataTypes:
+  - DeviceNetworkEvents
+tactics:
+- Command and control
+query: |
+  let excludedProcessFileNames = datatable (browser:string)["teams.exe","GoogleUpdate.exe","outlook.exe","msedge.exe","chrome.exe","iexplorer.exe","brave.exe","firefox.exe"]; //add additional browsers, mail clients, and legitimate google desktop applications where needed for exclusion in your environment
+  DeviceNetworkEvents 
+  | where not(InitiatingProcessFileName has_any (excludedProcessFileNames))
+  | where RemoteUrl has_any ("oauth2.googleapis.com","sheets.googleapis.com","drive.googleapis.com") and isnotempty(InitiatingProcessFileName)
+  | summarize visitedURLs=make_list(RemoteUrl) by ActionType, DeviceName, InitiatingProcessAccountName, InitiatingProcessParentFileName, InitiatingProcessFileName
+  | project ActionType, DeviceName, InitiatingProcessAccountName, InitiatingProcessParentFileName, InitiatingProcessFileName, visitedURLs, Connections=array_length(visitedURLs)
+  //| where visitedURLs contains "oauth2.googleapis.com" and visitedURLs has_any ("sheets.googleapis.com","drive.googleapis.com") // may allow for higher fidelity as the GC2 go application communicates to both the google drive folder and sheets API.
```

Comment:
Please map the entities as well. This would require extra extend statements.
Version is missing.

Reply:
Apologies for the delay. I had some personal things to attend to. I have amended the YAML and checked in VSCode. I have also added extend statements and field mappings as requested. I'll just check everything is looking good after the initial tests.